### PR TITLE
Feature/document variants

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Makes document workspace work with variants.

variant properties.
using variant data, if they are configured to do so.
a variant context API has been added.
variant selector now picks between variants. (still needs styling, I added a GH issue for it, up for grabs)
propaly even more things.

a screenshot here:
image

Notice, PR: https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/527 must be merged first!

### CODE REVIEW

**Comment by iOvergaard** [comment]
It looks like the Language selector was included in several sections: Media, Members, and Settings.

File: src/backoffice/settings/languages/workspace/language/views/edit/edit-language-workspace-view.element.ts [hunk 1]
```diff
@@ -3,10 +3,10 @@ import { UUITextStyles } from '@umbraco-ui/uui-css';
 import { css, html, nothing } from 'lit';
 import { customElement, state } from 'lit/decorators.js';
 import { ifDefined } from 'lit/directives/if-defined.js';
-import { UmbChangeEvent } from '@umbraco-cms/events';
 import { UmbLanguageWorkspaceContext } from '../../language-workspace.context';
 import { UmbLitElement } from '@umbraco-cms/element';
 import { LanguageModel } from '@umbraco-cms/backend-api';
+import { UmbChangeEvent } from '@umbraco-cms/events';
 import UmbInputCultureSelectElement from 'src/backoffice/shared/components/input-culture-select/input-culture-select.element';
 import UmbInputLanguagePickerElement from 'src/backoffice/shared/components/input-language-picker/input-language-picker.element';
```
Comment (iOvergaard) [inline src/backoffice/settings/languages/workspace/language/views/edit/edit-language-workspace-view.element.ts:11]: Use relative imports here

**Comment by iOvergaard** [comment]
I need to click two times on the document to get to the "Document" content app (not sure if this was caused by this PR):

2023-02-23 at 10 50 02 - Blush Earwig

File: src/backoffice/shared/components/workspace-property/workspace-property.element.ts [hunk 2]
```diff
@@ -213,9 +232,11 @@ export class UmbWorkspacePropertyElement extends UmbLitElement {
 		return html`
 			<umb-workspace-property-layout
 				id="layout"
+				alias="${ifDefined(this._alias)}"
 				label="${ifDefined(this._label)}"
 				description="${ifDefined(this._description)}">
 				${this._renderPropertyActionMenu()}
+				<p slot="description">${this._variantDisplayName}</p>
```
Comment (iOvergaard) [inline src/backoffice/shared/components/workspace-property/workspace-property.element.ts:239]: I get a warning here saying that slot=description does not exist. Perhaps missing @slot jsdoc?

**Review by iOvergaard — COMMENTED** [review]
Left a few questions but functionality wise it looks great.

**Comment by madsrasmussen** [comment]
> e

This is a known "bug". We will need a new extension point to show this on the correct section

**Comment by nielslyngsoe** [comment]
Hi @iOvergaard

The app-culture/language picker is part of the main branch, so not part of this PR.

The double-clicking is the router problem that I'm looking at as well.

I will have a look at the two issues :-)

File: src/backoffice/shared/components/workspace-property/workspace-property.element.ts (hunk 2 above)
Comment (nielslyngsoe) [reply src/backoffice/shared/components/workspace-property/workspace-property.element.ts:239]: That right, I have correct that now

File: src/backoffice/settings/languages/workspace/language/views/edit/edit-language-workspace-view.element.ts (hunk 1 above)
Comment (nielslyngsoe) [reply src/backoffice/settings/languages/workspace/language/views/edit/edit-language-workspace-view.element.ts]: I think this comes from main or Mads' PR.
I have corrected it in this branch.

**Review by iOvergaard — APPROVED** [review]
LGTM

**Pull request merged**